operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -279,6 +280,42 @@ func CreateOrUpdateLokiStack(
 		}
 	}
 
+	if stack.Spec.Replication != nil && len(stack.Spec.Replication.Zones) > 0 {
+		podList := &corev1.PodList{}
+		if err = k.List(ctx, podList, &client.ListOptions{
+			Namespace: stack.Namespace,
+			LabelSelector: labels.SelectorFromSet(labels.Set{
+				"app.kubernetes.io/instance": stack.Name,
+			}),
+		}); err != nil {
+			ll.Error(err, "Error getting pod resources")
+			return nil
+		}
+
+		for _, pod := range podList.Items {
+			assignedNode := &corev1.Node{}
+			key := client.ObjectKey{Name: pod.Spec.NodeName}
```

Comment:
I had a question about how Kubernetes treats the `topologySpreadConstraints` myself yesterday and tested out the error condition that I see the condition as a "debug information" for: picking a label for the `topologyKey` that does not actually exist on any Node.

What I wanted to find out was: does Kubernetes automatically modify the scheduling so that it only schedules on Nodes that have the required label(s). And the answer is: yes, it does.

While this is good news for our operator in general, as we do not need to add another `nodeAffinity` or something like that it means that this code will never be triggered, because the Pods will actually never be `scheduled` to a Node which doesn't have the matching labels. This means that `pod.Spec.NodeName` will be empty and the Pod will perpetually be in the `Pending` state.

Reply:
> If there are pending Pods, check if there's at least one Node that has all the required labels mentioned as topologyKey

If there are nodes that have matching labels, other pods will have already been scheduled to them until `maxSkew` is reached between the nodes.
My understanding is that a pending pod (reason: unschedulable) means that it cannot be scheduled even when there are nodes that have the topologyKey label